### PR TITLE
JVM argument tweaks

### PR DESCRIPTION
**Pull request by mikehearn**
Configure Capsule/Driver to give Corda a relatively poxy 200mb heap by default: we don't seem to need more for development purposes. Switch to G1GC by default and configure a default name for the VisualVM program to use.

### CODE REVIEW

File: tools/explorer/src/main/kotlin/net/corda/explorer/Main.kt [hunk 1]
```diff
@@ -107,7 +107,7 @@ class Main : App(MainView::class) {
 fun main(args: Array<String>) {
     val portAllocation = PortAllocation.Incremental(20000)
     driver(portAllocation = portAllocation) {
-        val user = User("user1", "test", permissions = setOf(startFlowPermission<CashFlow>()))
+        val user =  User("user1", "test", permissions = setOf(startFlowPermission<CashFlow>()))
```
Comment (Clintonio) [inline tools/explorer/src/main/kotlin/net/corda/explorer/Main.kt]: Extra space.

File: node/capsule/build.gradle [hunk 2]
```diff
@@ -49,15 +49,25 @@ dependencies {
...
+        // This version is known to work and avoids earlier 8u versions that have bugs.
+        minUpdateVersion['1.8'] = '102'
         caplets = ['CordaCaplet']
+
+        // JVM configuration:
+        // - Constrain to small heap sizes to ease development on low end devices.
+        // - Switch to the G1 GC which is going to be the default in Java 9 and gives low pause times/string dedup.
+        //
+        // If you change these flags, please also update Driver.kt
+        jvmArgs = ['-Xmx200m', '-XX:+UseG1GC']
```
Comment (Clintonio) [inline node/capsule/build.gradle]: Have you checked this is easy to override when running corda.jar?

Comment (mikehearn) [inline node/capsule/build.gradle]: You just specify it as normal. Capsule copies it across to the sub VM.

File: tools/explorer/src/main/kotlin/net/corda/explorer/Main.kt (hunk 1 above)
Comment (mikehearn) [inline tools/explorer/src/main/kotlin/net/corda/explorer/Main.kt]: Done

**Comment by mikehearn** [comment]
Ball in your court @Clintonio

File: docs/source/node-administration.rst [hunk 3]
```diff
@@ -56,9 +56,28 @@ Here are a few ways to build dashboards and extract monitoring data for a node:
...
   data to their service on a regular schedule.
 
+Memory usage and tuning
+-----------------------
+
+All garbage collected programs can run faster if you give them more memory, as they need to collect less
+frequently. As a default JVM will happily consume all the memory on your system if you let it, Corda is
+configured with a relatively small 200mb Java heap by default. When other overheads are added, this yields
+a total memory usage of about 500mb for a node (the overheads come from things like compiled code, metadata,
+off-heap buffers, thread stacks etc).
```
Comment (shamsasari) [inline docs/source/node-administration.rst]: comma after "stacks"

File: docs/source/node-administration.rst [hunk 4]
```diff
@@ -56,9 +56,28 @@ Here are a few ways to build dashboards and extract monitoring data for a node:
...
+
+If you want to make your node go faster and profiling suggests excessive GC overhead is the cause, or if your
+node is running out of memory, you can give it more by running the node like this:
+
+``java -Xmx1024m -jar corda.jar``
+
+The example command above would give a 1 gigabyte Java heap.
+
+.. note:: Unfortunately the JVM does not let you limit the total memory usage of Java program, just the heap size.
```
Comment (shamsasari) [inline docs/source/node-administration.rst]: of the Java program

**Pull request merged**